pkg/cloud/scope/machine.go

```diff
@@ -138,8 +139,9 @@ func (m *MachineScope) GetProviderID() string {
 }
 
 // SetProviderID sets the AWSMachine providerID in spec.
-func (m *MachineScope) SetProviderID(v string) {
-	m.AWSMachine.Spec.ProviderID = pointer.StringPtr(v)
+func (m *MachineScope) SetProviderID(instanceID, availabilityZone string) {
```

Comment:
can we put some units for this?

Reply:
it might seem trivial enough as to not deserve a test, but the func is actually relying in a handwritten fmt format and an external given implementation for `StringPtr`. It’s very cheap fo us to put a unit and failry valuable if some one considers to change the impl details later on. Also it makes way easier to visualise at a glange what provider should look like for the different cases.